java/ql/src/Stubs/MinimalStubsFromSource.ql

```diff
@@ -0,0 +1,27 @@
+/**
+ * Tool to generate java stubs from a qltest snapshot.
+ *
+ * It finds all declarations used in the source code,
+ * and generates minimal java stubs containing those declarations
+ * and their dependencies.
+ */
+
+import java
+import Stubs
+
+/** Declarations used by source code. */
+class UsedInSource extends GeneratedDeclaration {
```

Comment:
In C#, we're moving away from this minimal stub approach, which relies on the test input. Instead we're going to generate complete stubs of nuget packages. The stubbing API doesn't change, we're just using a different configuration: https://github.com/github/codeql/pull/5664/files#diff-fe12d2e3e080fe834b3b602ced8689b223d6785ee52c12b12268b413713e348a

Also, generating stubs for packages without extra filtering on types might help identify constructs that are not handled by the current stubbing implementation.

Reply:
> we're going to generate complete stubs of [...] packages

Definitely worth considering in the future, but this is currently not supported by the Java extractor, so it seems out of scope for this pull request.